### DISCUSSION TITLE
How does TAB completion with comma ',' work with zsh-autocomplete?

### DISCUSSION THREAD

**Discussion by jefftemplon** [marlonrichert/zsh-autocomplete#522] [General]
Hi,

The documentation says that zsh-autocomplete works with fasd.  The main feature I use from fasd is for commands like

`less ,2022120<TAB>` 

where fasd will look in its cache file for all files that contain `2022120` and present these to me as completion possibilities.  When `zsh-autocomplete` is enabled, trying this will instead attempt to tab-complete any matching files in the current working directory.   How can I get the desired `fasd` behavior?  Thanks

**Comment by marlonrichert** [comment]
Autocomplete currently doesn’t complete anything else from FASD than recent directories. Once #418 has been implemented, it will become possible to use FASD’s Tab widget instead of Autocomplete’s. 

However, it’s unlikely I will ever build in full support for FASD, since [that project has officially been abandoned](https://github.com/clvv/fasd). Of course, if anyone wants to support a PR for this, go ahead and I will have a look at it.

**Comment by jefftemplon** [comment]
Hi,

Thanks for the answer!  Yes, the abandonment.  That's why I came looking at zsh-autocomplete :) 

Maybe there is a different way to do it in zsh-autocomplete without fasd?  What I am looking for is a way to complete historical files based on a partial filename, so if I typed `202211` and used whatever the completion keystroke was, it would present as one of the  completion candidates a file `/var/lib/torque/server_logs/20221128` that I had accessed a couple days ago (despite my CWD being `/user/jtemplon` at that moment).  Is there something similar already supported in zsh-autocomplete?  If so I would happily use that.

**Reply by marlonrichert** [reply]
No, there currently isn’t. [FASD’s implementation](https://github.com/clvv/fasd/blob/90b531a5daaa545c74c7d98974b54cbdb92659fc/fasd#L480), though, looks fairly simple. So, feel free to submit a pull request with that.

However, please read #404 first.

**Reply by jefftemplon** [reply]
Hi,

Implementing it myself in zsh-autocomplete is a bit outside of my scope and range of shell-coding powers. I will play around with the pre-existing stuff and see what works!
